App.tsx

```diff
@@ -66,33 +77,36 @@ const App = () => {
   }, []);
 
   // callback to get splashScreen animation completion
-  const screenAnimationComplete = (animation) => {
+  const screenAnimationComplete = useCallback((animation) => {
     setSplashAnimation(animation);
-  };
+  }, []);
+
+  let body = <SplashScreen screenAnimationComplete={screenAnimationComplete} />;
```

Comment:
We could have used `setSplashAnimation` directly here, but did not do so because in the future if we want to add more actions on the completion of a screenAnimation, then it becomes easier to make the change then. In short the useCallback provides both flexibility and minimises un-necessary re-renders